packages/zksync-dai-bridge/contracts/l1/L1DAITokenBridge.sol

```diff
@@ -0,0 +1,216 @@
+    address _from,
+    address _to,
+    uint256 _value
+  ) external returns (bool success);
+}
+
+interface L2DAITokenBridgeLike {
+  function finalizeDeposit(
+    address _from,
+    address _to,
+    address _l1Token,
+    uint256 _amount,
+    bytes calldata _data
+  ) external;
+}
+
+// Managed locked funds in L1Escrow and send / receive messages to L2DAITokenBridge counterpart
+// Note: when bridge is closed it will still process in progress messages
+
+contract L1DAITokenBridge is IL1Bridge {
+  // --- Auth ---
+  mapping(address => uint256) public wards;
+
+  function rely(address usr) external auth {
+    wards[usr] = 1;
+    emit Rely(usr);
+  }
+
+  function deny(address usr) external auth {
+    wards[usr] = 0;
+    emit Deny(usr);
+  }
+
+  modifier auth() {
+    require(wards[msg.sender] == 1, "L1DAITokenBridge/not-authorized");
+    _;
+  }
+
+  uint256 constant DEPOSIT_ERGS_LIMIT = 2097152;
+  address constant BOOTLOADER_ADDRESS = 0x0000000000000000000000000000000000008001; // address(SYSTEM_CONTRACTS_OFFSET + 0x01);
+  address public immutable l1Token;
+  address public immutable l2Bridge;
+  address public immutable l2Token;
+  address public immutable escrow;
+  IMailbox public immutable zkSyncMailbox;
+  uint256 public isOpen = 1;
+
+  mapping(uint256 => mapping(uint256 => bool)) public isWithdrawalFinalized;
+  mapping(address => mapping(bytes32 => uint256)) depositAmount;
+
+  event Rely(address indexed usr);
+  event Deny(address indexed usr);
+  event Closed();
+
+  constructor(
+    address _l1Token,
+    address _l2DAITokenBridge,
+    address _l2Token,
+    address _escrow,
+    IMailbox _mailbox
+  ) {
+    wards[msg.sender] = 1;
+    emit Rely(msg.sender);
+
+    l1Token = _l1Token;
+    l2Bridge = _l2DAITokenBridge;
+    l2Token = _l2Token;
+    escrow = _escrow;
+    zkSyncMailbox = _mailbox;
+  }
+
+  function l2TokenAddress(address _l1Token) public view returns (address) {
+    require(_l1Token == l1Token, "L1DAITokenBridge/token-not-dai");
+    return l2Token;
+  }
+
+  function close() external auth {
+    isOpen = 0;
+
+    emit Closed();
+  }
+
+  //TODO: Check if reentrancy guard is needed for all these methods
+  function deposit(
+    address _l2Receiver,
+    address _l1Token,
+    uint256 _amount
+  ) external payable returns (bytes32 txHash) {
+    require(_l1Token == l1Token, "L1DAITokenBridge/token-not-dai");
+    require(isOpen == 1, "L1DAITokenBridge/closed");
+
+    TokenLike(l1Token).transferFrom(msg.sender, escrow, _amount);
+
+    bytes memory l2TxCalldata = abi.encodeWithSelector(
+      L2DAITokenBridgeLike.finalizeDeposit.selector,
+      msg.sender,
+      _l2Receiver,
+      _l1Token,
+      _amount,
+      "" // _data is not used in this bridge but still kept as an argument of finalizeDeposit for consistency with ZkSync's standard ERC20 bridge
+    );
+
+    txHash = zkSyncMailbox.requestL2Transaction{value: msg.value}(
+      l2Bridge,
+      0, // l2Value is always 0
```

Comment:
This represents the msg.value to send on L2, which in the case of the `finalizeDeposit` and `relay` L2 methods, should be 0. Note that the Dai amount itself is encoded in `l2TxCalldata`.

Reply:
Why is this value always zero for `finalizeDeposit` and `relay` though? I don't see it in the documentation.

Reply:
Because these L2 methods aren't `payable`. This is identical to what we've done for the arbitrum dai bridge: https://github.com/makerdao/xdomain/blob/master/packages/arbitrum-dai-bridge/contracts/l1/L1CrossDomainEnabled.sol#L52

Reply:
Ah I see. Okay cool.